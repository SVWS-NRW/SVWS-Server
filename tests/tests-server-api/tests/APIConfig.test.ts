import { describe, expect, test } from "vitest";
import { getApiServer } from "./utils/TestUtils.js";
import { DBSchemaListeEintrag } from "@core";

const allowDestructiveTests = process.env.MODE === 'allowDestructiveTests'

describe("Config Tests", () => {
	describe.each([{schema: "GymAbi01"}])('gegen %s', ({schema}) => {
		const api = getApiServer(schema);

		test.skip("setClientConfigGlobalKey", async () => {
			// const result = await api.setClientConfigGlobalKey(schema, 1)
			// expect(result).toBeTruthy();
		});

		test.skip("setClientConfigUserKey", async () => {
			// const result = await api.setClientConfigUserKey(schema, 1)
			// expect(result).toBeTruthy();
		});

		test("getConfigCertificate", async () => {
			const result = await api.getConfigCertificate();
			expect(result).toBeDefined();
		});

		test("getConfigCertificateBase64", async () => {
			const result = await api.getConfigCertificateBase64();
			expect(result).toBeDefined();
		});

		test.runIf(allowDestructiveTests)("getConfigDBSchemata: hole Liste mit DBs", async () => {
			const result = await api.getConfigDBSchemata();
			expect(result).toMatchSnapshot();
			expect(result.toArray()[0]).toBeInstanceOf(DBSchemaListeEintrag);
		});

		test("getConfigPublicKeyBase64", async () => {
			const result = await api.getConfigPublicKeyBase64();
			expect(result).toBeDefined();
		});
	})
})
