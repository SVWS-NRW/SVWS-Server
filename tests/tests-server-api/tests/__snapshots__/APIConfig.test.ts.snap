// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Config Tests > gegen { schema: 'GymAbi01' } > getConfigDBSchemata: hole Liste mit DBs 1`] = `
ArrayList {
  "addFirst": [Function],
  "addLast": [Function],
  "elementData": [
    DBSchemaListeEintrag {
      "isDefault": true,
      "name": "GymAbi01",
    },
  ],
  "getFirst": [Function],
  "getLast": [Function],
  "modCount": 1,
  "removeFirst": [Function],
  "removeLast": [Function],
}
`;
